docs/guides/Package-authoring-best-practices.md

```diff
@@ -0,0 +1,177 @@
+---
+title: Package authoring best practices
+description: A general guide of best practices for creating high quality NuGet packages.
+author: chgill-MSFT
+ms.author: chgill
+ms.date: 09/17/20
+ms.topic: conceptual
+---
+
+# Package authoring best practices
+
+This guidance is intended to give NuGet.org package authors a light-weight end to end reference for creating and publishing high quality packages. It will primarily focus on package-specific best practices such as metadata and packing. For more in depth suggestions for building high quality libraries, see the .NET [Open-source library guidance](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/).
```

Comment:
Is `<frameworkAssemblies>` even interesting anymore? Should we have an AVOID? I've always felt that I only want a project to bring in project references and I'll select my direct assembly references myself.

Reply:
I think it's still relevant. Even though most of the BCL is available as packages now, by default a project targeting .NET Framework will not reference `System.Net.Http`, so a customer wanting to use `HttpClient` will either need to reference the assembly, or the package. Referencing the package requires a restore, which is not needed for an assembly reference. Referencing the package introduces all sorts of risks of package downgrade warnings, and I've seen some assembly version issues as well (MSBuild warnings in Resolve Assembly References), whereas assembly references seem to have fewer issues.

Or maybe this is just my personal preference.  But to me, as long as .NET Framework is in use, I will always prefer assembly references over packages.

Reply:
Okay, sounds good! Maybe this would be a good "CONSIDER" then 😄